pkg/controller/node/node_controller_test.go

```diff
@@ -0,0 +1,101 @@
+package node
+
+import (
+	"context"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes/scheme"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	nmstatev1alpha1 "github.com/nmstate/kubernetes-nmstate/pkg/apis/nmstate/v1alpha1"
+)
+
+var _ = Describe("Node controller reconcile", func() {
+	var (
+		cl         client.Client
+		reconciler ReconcileNode
+		request    reconcile.Request
+		nodeName   = "node01"
+		node       = corev1.Node{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: nodeName,
+				UID:  "12345",
+			},
+		}
+		nodenetworkstate = nmstatev1alpha1.NodeNetworkState{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: nodeName,
+			},
+		}
+	)
+	BeforeEach(func() {
+		s := scheme.Scheme
+		s.AddKnownTypes(nmstatev1alpha1.SchemeGroupVersion,
+			&nmstatev1alpha1.NodeNetworkState{},
+		)
+
+		objs := []runtime.Object{&node, &nodenetworkstate}
+
+		// Create a fake client to mock API calls.
+		cl = fake.NewFakeClientWithScheme(s, objs...)
+
+		reconciler.client = cl
+	})
+	Context("when node is not found", func() {
+		BeforeEach(func() {
+			request.Name = "not-present-node"
```

Comment:
Would you please initialize the whole request in this BeforeEach? So it is clear what is a part of it and that we are not reusing previously popoulated object.

Reply:
There is nothing to initialize there.

Reply:
You are setting stuff in the shared global variable.l, that's not really transparent. I was asking to do request = Request{Name: ...} here